apps/pkcs12.c

```diff
@@ -838,6 +868,29 @@ int pkcs12_main(int argc, char **argv)
     return ret;
 }
 
+static int jdk_trust(PKCS12_SAFEBAG *bag, void *cbarg)
+{
+    STACK_OF(X509_ATTRIBUTE) *attrs = NULL;
+    X509_ATTRIBUTE *attr = NULL;
+
+    /* Nothing to do */
+    if (!cbarg)
+        return 1;
+
+    /* Get the current attrs */
+    attrs = PKCS12_SAFEBAG_get0_attrs(bag);
+
+    /* Create a new attr for the JDK Trusted Usage and add it */
+    attr = X509_ATTRIBUTE_create(NID_oracle_jdk_trustedkeyusage, V_ASN1_OBJECT, (ASN1_OBJECT*)cbarg);
+    X509at_add1_attr(&attrs, attr);
```

Comment:
Maybe add a comment saying that if `attrs` is still NULL it will be initialised here.